verify/helpers_test.go

```
@@ -0,0 +1,200 @@
+			profileStatus: "BLOCKED",
+			chaincode:     "abc123",
+			expectedErr:   errors.New("profile url is not a string"),
+		},
+		{
+			name:          "missing chaincode",
+			userId:        "3",
+			profileURL:    "http://test.com",
+			profileStatus: "BLOCKED",
+			chaincode:     "",
+			expectedErr:   errors.New("chaincode is blocked"),
+		},
+		{
+			name:          "invalid chaincode",
+			userId:        "4",
+			profileURL:    "http://test.com",
+			profileStatus: "BLOCKED",
+			chaincode:     123,
+			expectedErr:   errors.New("chaincode is not a string"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			client.Collection("users").Doc(testCase.userId).Set(ctx, map[string]interface{}{
+				"chaincode":     testCase.chaincode,
+				"profileURL":    testCase.profileURL,
+				"profileStatus": testCase.profileStatus,
+			})
+
+			profileURL, profileStatus, chaincode, err := getUserData(client, ctx, testCase.userId)
+
+			assert.Equal(t, testCase.expectedErr, err)
+			if testCase.expectedErr == nil {
+				assert.Equal(t, testCase.profileURL.(string), profileURL)
+				assert.Equal(t, testCase.profileStatus.(string), profileStatus)
+				assert.Equal(t, testCase.chaincode.(string), chaincode)
+			}
+		})
+	}
+}
+
+func TestGetUserIdFromBody(t *testing.T) {
+	// valid request body
+	body := []byte(`{"userId": "123"}`)
+	expected := "123"
+
+	actual := getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+
+	// empty request body
+	body = []byte(``)
+	expected = ""
+
+	actual = getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+
+	// invalid request body
+	body = []byte(`{"invalidProperty": ""}`)
+	expected = ""
+
+	actual = getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+}
```

i guess this should also throw an error if an invalid property is passed.

---

It is not related to unmarshal but I am talking about API behaviour

---

Please create a ticket for this as discussed

---

**Created Ticket:** https://github.com/Real-Dev-Squad/identity-service/issues/145